File: src/resources/drift.ts
import type { HttpClient } from '../http.js';
import type { DriftCatch, DriftComparison, DriftComparisonQuery } from '../types/drift.js';

export class DriftResource {
  constructor(private http: HttpClient) {}

  async catches(): Promise<DriftCatch[]> {
    return this.http.get<DriftCatch[]>('/v1/drift/catches');
  }

  async comparison(opts?: DriftComparisonQuery): Promise<DriftComparison> {
    return this.http.get<DriftComparison>('/v1/drift/comparison', {
      params: opts as Record<string, string | undefined>,
    });
  }
}

File: packages/python/invariance/resources/drift.py
from __future__ import annotations

from typing import Any

from ..http_client import HttpClient
from ..types import DriftCatch, DriftComparison, DriftComparisonQuery


class DriftResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def catches(self) -> list[DriftCatch]:
        return await self._http.get("/v1/drift/catches")

    async def comparison(
        self, opts: DriftComparisonQuery | None = None
    ) -> DriftComparison:
        return await self._http.get("/v1/drift/comparison", params=opts)
